sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -776,7 +791,7 @@ public List<EventProcessor> getEventProcessors() {
    */
   @Override
   public void addEventProcessor(final @NotNull EventProcessor eventProcessor) {
-    eventProcessors.add(eventProcessor);
+    eventProcessors.add(new EventProcessorAndOrder(eventProcessor, eventProcessor.getOrder()));
```

Comment:
I might be missing something, but what's the point of having a `EventProcessorAndOrder` class when the order value is anyway always present in `eventProcessor.getOrder()`?

Reply:
`getOrder` may return `null` defaulting to current nano time, which should mean it goes after all event processors currently in place, except for those with an actual number that is higher. Those would be intended to go last and probably have order close to `Long.MAX_VALUE`.

Reply:
Also see the ctor of `EventProcessorAndOrder`